omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,242 @@
+		description.addRequiredProperty("identifierType");
+		description.addProperty("location");
+		description.addRequiredProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = getCreatableProperties();
+		description.removeProperty("identifierType");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		Object identifierTypeUuid = postBody.get("identifierType");
+		ArrayList<String> errors = new ArrayList<String>();
+		IdentifierSource identifierSource = null;
+		PatientIdentifierType patientIdentifierType = null;
+		if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+			errors.add("source");
+		} else {
+			identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+			if (identifierSource == null) {
+				errors.add("source");
+			}
+		}
+		if (identifierTypeUuid == null || StringUtils.isBlank(identifierTypeUuid.toString())) {
+			errors.add("identifierType");
+		} else {
+			patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+			if (patientIdentifierType == null) {
+				errors.add("IdentifierType");
+			}
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException(
+			        "The values of the following inputs are missing or invalid : " + errors.toString());
+		}
+		Location location = null;
+		if (locationUuid != null && StringUtils.isNotBlank(locationUuid.toString())) {
+			location = Context.getLocationService().getLocationByUuid(locationUuid.toString());
+		}
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		autoGenerationOption.setSource(identifierSource);
+		autoGenerationOption.setIdentifierType(patientIdentifierType);
+		autoGenerationOption.setManualEntryEnabled(this.parseBoolean(manualEntryEnabled));
+		autoGenerationOption.setAutomaticGenerationEnabled(this.parseBoolean(automaticallyEnabled));
+		autoGenerationOption.setLocation(location);
+		return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.FULL);
+	}
+	
+	@Override
+	public SimpleObject update(String uuid, SimpleObject postBody, RequestContext context) throws ResponseException {
+		int id = 0;
+		try {
+			id = Integer.parseInt(uuid);
+		}
+		catch (NumberFormatException e) {
+			throw new ValidationException("The uuid " + e.getMessage() + "can not be converted to an integer");
```

Comment:
This message can be a bit confusing when one does not provide a uuid and then it says that it (what they have not provided) cannot be converted to an integer.

Reply:
good point, but can someone do a PUT request without providing a Uuid?

Reply:
Yes it happens